From f826bf04a349ecc8ffb528cfea540b328809614c Mon Sep 17 00:00:00 2001
From: Phong Vo <pvo@apm.com>
Date: Mon, 16 Nov 2015 18:23:45 +0700
Subject: pci:xgene: hide X-Gene BAR0/1 in case of xgene_raw_pci_write

Xgene BAR0/1 of bridge should be hidden during enumeration to avoid
the sizing and resource allocation by PCIe core (refer to
xgene_pcie_hide_rc_bars). This had been aready been implemented
in xgene_raw_pci_read() but not xgene_raw_pci_write(). This patch
adds support for hiding X-Gene BAR0/1 in the case of xgene_raw_pci_write().

Signed-off-by: Chuong Tran <cltran@apm.com>
Signed-off-by: Phong Vo <pvo@apm.com>
---
 drivers/pci/host/pci-xgene.c | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/drivers/pci/host/pci-xgene.c b/drivers/pci/host/pci-xgene.c
index d4601e3..d7449b0 100644
--- a/drivers/pci/host/pci-xgene.c
+++ b/drivers/pci/host/pci-xgene.c
@@ -612,8 +612,16 @@ static int xgene_raw_pci_write(struct pci_mmcfg_region *cfg, unsigned int bus,
 	void __iomem *addr;
 	u32 mask, tmp;
 
-	if (bus == cfg->start_bus && devfn != 0)
-		return PCIBIOS_DEVICE_NOT_FOUND;
+	if (bus == cfg->start_bus) {
+		if (devfn != 0)
+			return PCIBIOS_DEVICE_NOT_FOUND;
+
+		/* see xgene_pcie_hide_rc_bars() above */
+		if (offset == PCI_BASE_ADDRESS_0 ||
+		    offset == PCI_BASE_ADDRESS_1) {
+			return PCIBIOS_SUCCESSFUL;
+		}
+	}
 
 	__set_rtdid_reg(cfg, bus, devfn);
 	addr = __get_cfg_base(cfg, bus) + (offset & ~0x3);
-- 
2.6.0

